src/Paprika/Store/PagedDb.cs

```diff
@@ -506,16 +504,88 @@ public IDictionary<Keccak, uint> IdCache
             }
         }
 
-        public void Prefetch(DbAddress address) => db.Prefetch(address);
+        public void Prefetch(DbAddress addr) => db.Prefetch(addr);
 
         public Page GetAt(DbAddress address) => db._manager.GetAt(address);
 
         public override string ToString() => $"{nameof(ReadOnlyBatch)}, Name: {name}, BatchId: {BatchId}";
     }
 
+    /// <summary>
+    /// A linked batch allows for an arbitrary depth nesting of transactions and committing them bottom up.
+    /// </summary>
+    /// <remarks>
+    /// As the mapping are stored in the page table, the linked batch allows to query the data with
+    /// one dictionary lookup and a fallback to the database.
+    /// 
+    /// Additionally, it can be built with other linked batches as dependencies and copy from their page tables.
+    /// The crux here is that only the previous one requires visiting as it has the copies of the page mapping
+    /// from the previous one. As the pages are overwritten when needed, the dictionary should not be too big. 
+    /// </remarks>
+    private class LinkedBatch : Batch
+    {
+        private static readonly UIntPtr WriteMarker = new(1UL << (UIntPtr.Size == 4 ? 31 : 63));
+        private static readonly UIntPtr AddressMask = ~WriteMarker;
+
+        private readonly Dictionary<DbAddress, UIntPtr> _pageTable = new();
```

Comment:
The max size would be the max reorg size for the given blockchain. This would be usually something like 128 even for the deepest reorgs and yes, this could be taken as a config parameter. Basically, if this is breached, there's something terribly wrong with the whole network finality.